tests/ui/privacy/private-in-public-warn.rs

```diff
@@ -147,7 +146,9 @@ mod impls_generics {
         pub fn f(arg: Priv) {} // OK
     }
     impl Pub<Priv> {
-        pub fn f(arg: Priv) {} // OK
+        pub fn f(arg: Priv) {}
+        //~^ WARNING type `impls_generics::Priv` is more private than the item `impls_generics::Pub::<impls_generics::Priv>::f`
+        //~| WARNING type `impls_generics::Priv` is more private than the item `impls_generics::Pub::<impls_generics::Priv>::f`
```

Comment:
Why is this a warning now?
Aren't we using non-shallow impl visiting for these lints?

Reply:
For associated items shallow version was used. I will split bug fix from this PR after https://github.com/rust-lang/rust/pull/113161 merge.